### DISCUSSION TITLE
Terminating an Instance is not Working

### DISCUSSION THREAD

**Discussion by Kemprateek03** [Azure/azure-functions-durable-extension#2429] [Q&A] [answered]
Hi, 
I'm trying to find all the instances that are running and want to terminate them.
FYI, I'm using an eternal orchestrator. 
My requirement is to terminate the Instance ID associated with the eternal orchestration and restart the eternal orchestrator with new set of input values.

Here is the image for reference - 
![image](https://user-images.githubusercontent.com/19287198/227714450-0003f5a7-1c3d-478a-bb54-bf06bc3caec6.png)

The above line of code is not terminating the Instance ID associated with the eternal orchestrator, instead the eternal orchestrator is running over and over.

**Comment by davidmrdavid** [comment]
Hi @Kemprateek03:

Thanks for reaching out. It would help us if you could provide us a bit more information on this issue.

Is this happening locally, or on Azure?
If this is happening locally, can you please provide us with a minimal reproducer?
If it's happening on Azure, could you please provide us with the following identifying information?:

- **Timeframe issue observed**:
- **Function App name**:
- **Function name(s)**:
- **Azure region**:
- **Orchestration instance ID(s)**:
- **Azure storage account name**:

Thanks!

**Reply by Kemprateek03** [reply, marked as answer]
Hey David, I got the fix for this solution.
It seems I was missing await while calling the 
client.terminate(instance_id,reason).

The Microsoft documentation seemed to just have client.terminate(instance_id,reason), instead of 
await client.terminate(instance_id,reason)  which caused this issue. 

Thanks for your response and appreciate it.
